chat-android/src/main/java/com/ably/chat/RoomStatus.kt

```diff
@@ -118,3 +162,60 @@ data class RoomStatusChange(
      */
     val error: ErrorInfo? = null,
 )
+
+open class RoomStatusEvenEmitter : EventEmitter<RoomLifecycle, RoomStatus.Listener>() {
+
+    override fun apply(listener: RoomStatus.Listener?, event: RoomLifecycle?, vararg args: Any?) {
+        try {
+            listener?.roomStatusChanged(args[0] as RoomStatusChange)
+        } catch (t: Throwable) {
+            Log.e("RoomEventEmitter", "Unexpected exception calling Room Status Listener", t)
+        }
+    }
+}
+
+class DefaultStatus(private val logger: LogHandler?) : InternalRoomStatus, RoomStatusEvenEmitter() {
```

Comment:
It's not Kotlin idiomatic to use inheritance here. It's much better to use composition

Reply:
Fixed https://github.com/ably-labs/ably-chat-kotlin/pull/27/commits/ce9c0baf4085c8199771f89c29b3d60fe399db26